units/commit_queue.go

```diff
@@ -74,12 +74,26 @@ func (j *commitQueueJob) TryUnstick(ctx context.Context, cq *commitqueue.CommitQ
 	}
 
 	// unstuck the queue if the patch is done.
-	if nextItem.Source == commitqueue.SourceDiff && !patch.IsValidId(nextItem.Issue) {
+	version := ""
+	if nextItem.Source == commitqueue.SourceDiff {
+		version = nextItem.Issue
+	} else if nextItem.Source == commitqueue.SourcePullRequest {
+		version = nextItem.Version
+	} else {
+		grip.Error(message.Fields{
+			"message": "stuck commit queue entry has unknown source",
+			"entry":   nextItem,
+			"project": projectRef.Identifier,
+			"job_id":  j.ID(),
+		})
+	}
+	if version == "" || !patch.IsValidId(nextItem.Issue) {
```

Comment:
this should check if patch.IsValidId(version) I believe; otherwise for PR items this will still always fail

Reply:
good catch, fixed